src/makefiles/default_rules.mk

```diff
@@ -50,7 +50,7 @@ $(LIBFILE): $(LIBNAME).a
 	ln -sf $(shell pwd)/$@ $(KALDILIBDIR)/$@
   else ifeq ($(shell uname), Linux)
         # Building shared library from static (static was compiled with -fPIC)
-	$(CXX) -shared -o $@ -Wl,--no-undefined -Wl,--as-needed  -Wl,-soname=$@,--whole-archive $(LIBNAME).a -Wl,--no-whole-archive $(LDFLAGS) $(LDLIBS)
+	$(CXX) -shared -o $@ -Wl,--as-needed  -Wl,-soname=$@,--whole-archive $(LIBNAME).a -Wl,--no-whole-archive $(LDFLAGS) $(LDLIBS)
```

Comment:
"--no-undefined: It reports the unresolved symbols that are not resolved immediately, at linking stage."

---

is not suggested to commonly use as here. We can control all needed libs are listed in link cmd. However, we cannot require third-party packages. As the openfst depends on automake, and libtool has an issue with a shared lib that links to another shared lib. So any target link with this openfst shared library would have build error like: "fst.h:746: undefined reference to `fst::FstHeader::Write"

Reply:
The no-undefined flag also is not help for optimized the lib sizes, that is the intension when it was added in one pull.